CMakeLists.txt

```diff
@@ -225,11 +225,8 @@ zephyr_compile_options(
 toolchain_ld_base()
 
 if(NOT CONFIG_NATIVE_APPLICATION)
-zephyr_ld_options(
-  -nostdlib
-  -static
-  -no-pie
-)
+  # @Intent: Set linker specific flags for bare metal target
+  toolchain_ld_baremetal()
```

Comment:
is qemu_x86 for instance "baremetal"?

Reply:
Yes because there Zephyr is the only/owning OS. Unlike Zephyr hosted under POSIX/Linux.
(The qemu part is irrelevant, as qemu_x86 means full-system simulation, not the user-emulation variant of qemu)

Reply:
I think you're saying that the toolchain part of the build doesn't and shouldn't care about qemu because it's located at a lower and different layer. Correct? If true then great but still confusing to see "baremetal" being used for qemu too considering qemu is all over the (other parts of the) documentation and a core piece of the validation. In every other example I've seen "bare + metal" is used to strongly negate any sort of emulation whatsoever, much stronger than just a local, "not posix" sense.

Reply:
Yes, because from the build system's point of view it builds an ELF that for all intents and purposes is intended for real hardware. That we then cheat it by actually running the ELF on a simulated target, is a concern we shouldn't complicate the build system with.

I'm open to alternate naming. Is `toolchain_ld_not_posix` more readable?

Reply:
Some targets have some POSIX elements so "native" is more accurate, it's also more "standard" https://crosstool-ng.github.io/docs/toolchain-types/

I don't like negative names either because they're one step away from double negations (e.g., `if (not disabled)`, `no -nostdlib`, etc.), yet in this particular case `toolchain_ld_not_native` is the best I can see so far. Maybe `toolchain_ld_not_native_posix` if you don't mind the length.
https://docs.zephyrproject.org/latest/boards/posix/native_posix/doc/board.html